src/systemdhook.c

```diff
@@ -570,26 +478,39 @@ static int prestart(const char *rootfs,
 		}
 	}
 
-	if (!contains_mount(config_mounts, config_mounts_len, "/sys/fs/cgroup")) {
-		/* libmount */
-		struct libmnt_table *tb = NULL;
-		int rc = -1;
+	if (!contains_mount(config_mounts, config_mounts_len, "/sys/fs/cgroup/systemd")) {
+		_cleanup_free_ char *dest = NULL;
+		rc = asprintf(&dest, "%s/%s", rootfs, "/sys/fs/cgroup");
+		_cleanup_free_ char *systemd_dest = NULL;
+		rc = asprintf(&systemd_dest, "%s/%s", dest, "systemd");
 
-		/*
-		 * initialize libmount
-		 */
-		mnt_init_debug(0);
+		if ((umount(dest) == -1) && errno != EINVAL) {
+			pr_perror("Failed to umount %s", dest);
+		}
 
-		tb = parse_tabfile("/proc/self/mountinfo");
-		if (!tb) {
+		/* Mount tmpfs at new temp directory */
```

Comment:
Not required since the docker cgroup mount takes care of this already. (Maybe add a check for runc?)